docs/extensibility/gem-builder/gem-builder-ui.md

```diff
@@ -0,0 +1,5128 @@
+---
+title: Build Gem UI
+id: gem-builder-ui
+slug: /engineers/gem-builder-ui
+description: Build the UI of your custom gem using uispec.py classes
+tags:
+  - gem builder
+---
+
+In Prophecy, the `dialog()` function defines the visual configuration interface for your gem. This is what users see and interact with when setting up or editing a gem in the UI.
+
+To build an effective dialog, you'll use layout containers, UI controls (like checkboxes, text boxes, expression editors), and bindings to connect UI elements to your gem's configuration. All of these elements are imported into your custom gem from `uispec.py`.
+
+By examining `uispec.py`, you can:
+
+- Discover the full library of UI elements.
+- Understand what each component does and what methods it supports.
+- See how to configure visual behavior, bind properties, add hints/tooltips, and more.
+
+:::info
+To see an example `dialog()` function, visit [Gem Builder reference](/engineers/gem-builder-reference#dialog-ui).
+:::
+
+## Atoms
+
+Atoms are the basic UI components used in the gem dialog configuration. Here are some common Atoms found in `uispec.py`:
+
+| Atom Class      | Description                                               |
+| --------------- | --------------------------------------------------------- |
+| `Checkbox`      | Boolean toggle control                                    |
+| `Switch`        | Boolean toggle with a different visual style              |
+| `TextBox`       | Simple one-line text input                                |
+| `ExpressionBox` | Input for code expressions, often with schema suggestions |
+| `ExpTable`      | Table where users can add rows of expressions             |
+| `BasicTable`    | Configurable table with editable cells                    |
+| `TitleElement`  | Static title or heading text                              |
+| `AlertBox`      | Message box for alerts/info/warnings                      |
+| `Button`        | Button element that can trigger actions                   |
+| `FileSelector`  | UI for selecting files or folders                         |
+| `MultiSelect`   | Dropdown with multiple selection support                  |
```

Comment:
Let's add

TextBox, SColumn, SColumnExpression, TextArea, SchemaTable, Ports, Schema Column dropdown, SelectBox, Markdown, RadioGroup, Editor

and in another table let's add all the layout ones like:
Stack layout, column layout, field picker

Also let's remove multiselect, fileselector from list (there is no such element)